### ISSUE TITLE
Where is the resize drag handle gone in jupyter?

### ISSUE THREAD

**Issue by leolchat** [bokeh/bokeh#7793]
I used to be able to resize a figure in jupyter notebook (the grey handle at the bottom right of the figure [here](http://i.imgur.com/nqJo2Jn.png).
But It seems to not be on my system anymore, same with jupyter-lab.

Software versions are all latest from pip on ubuntu 18.04 so:
BokehJS 0.12.14
matplotlib 2.2.2
ipywidgets 7.1.2
jupyterlab_bokeh v0.4.0

**Comment by bryevdv** [comment]
As I said, the bokeh resize tool was removed years ago, it did not exist any longer. That is why there is no documentation on it.

**Comment by bryevdv** [comment]
Also, again: the image you linked is not of a bokeh plot. You may very well have better luck if you ask in a more appropriate place

**Comment by bryevdv** [comment]
Or, if your are asking if there is any way to get the MPL resize handle to work on a bokeh plot, then the answer is no, they have never had any relation or usage together.

**Comment by leolchat** [comment]
ok, well thank you anyway. I am trying on jupyter and matplotlib. It is just not clear in the stack of software who is responsible to create that resize.
Reworded, my question would be If jupyter notebook/lab provided it, it would just work for bokeh?

**Comment by bryevdv** [comment]
No, the tool is a party of the plotting library (in your case MPL) not something that the notebook would be responsible for.

**Comment by leolchat** [comment]
Is bokeh using MPL as a backend? Why is 'my case' MPL? I am using bokeh if that is not clear. I just found that picture because that what it looked like in my memory when I was using bokeh a year ago or so.

**Comment by leolchat** [comment]
So I guess we could change that issue into a feature request?

**Comment by bryevdv** [comment]
> Is bokeh using MPL as a backend? Why is 'my case' MPL? I am using bokeh if that is not clear. 

It was definitely not clear. The image you linked, which I assumed was relevant to the question, is not of a Bokeh plot. The controls it shows are the controls that show up when MPL renders in notebooks. Absent more context, or an explanation of why you would link to an example image of an MPL plot in the Bokeh tracker, it seemed safe to assume you'd simply made a mistake of some sort. 

> Is bokeh using MPL as a backend?

Bokeh does not use MPL in any capacity. MPL generates static images (PNG, SVG, etc) directly from Python. Bokeh generates a declarative specification of a plot (or of a larger data app) and sends that to be rendered as live content by a browser by the BokehJS library (which is what supports all the interactive capabilities unique to Bokeh).

As for making this a feature request: the resize tool was explicitly removed on purpose for a reason, not capriciously. It was not compatible with other layout capabilities that were deemed more important than it. Unfortunately I don't see a path to adding it back as a generally available tool any time soon.

**Comment by bryevdv** [comment]
If the resize tool is crucial to your work, I will mention that is seems to have been removed in `0.12.7` so releases prior to that should still have it. 

Alternatively Bokeh is extensible. Although a completely general resize tool is not compatible with the general layout system (which is why it would not be appropriate to add it to the core library), it might be possible to create a [custom tool extension] that works in some limited cases (i.e. your particular use case)

**Comment by leolchat** [comment]
When poking around, I was hoping that a layout would be able to handle that, at least if it is at the top level. But even `sizing_mode='scale_width'` doesn't update when the window is resized :/ So I was looking elsewhere.
